samcli/commands/build/build_context.py

```diff
@@ -64,10 +64,14 @@ def __init__(
         self._container_manager: Optional[ContainerManager] = None
 
     def __enter__(self) -> "BuildContext":
-        self._template_dict = get_template_data(self._template_file)
 
-        self._function_provider = SamFunctionProvider(self._template_dict, self._parameter_overrides)
-        self._layer_provider = SamLayerProvider(self._template_dict, self._parameter_overrides)
+        stacks = SamBuildableStackProvider.get_buildable_stacks(
+            self._template_file, parameter_overrides=self._parameter_overrides
+        )
+        self._template_dict = SamBuildableStackProvider.find_root_stack(stacks).template_dict
```

Comment:
if the template is not correct I remember it will raise exception before getting here. Like this: amcli.commands.validate.lib.exceptions.InvalidSamDocumentException